phys/module_sf_sfclay.F

```diff
@@ -241,14 +241,14 @@ SUBROUTINE SFCLAY(U3D,V3D,T3D,QV3D,P3D,dz8w,                    &
                 ids,ide, jds,jde, kds,kde,                         &
                 ims,ime, jms,jme, kms,kme,                         &
                 its,ite, jts,jte, kts,kte                          &
-#if defined(mpas) 
+#if defined(mpas)
                 ,isftcflx,iz0tlnd,scm_force_flux,                  &
                 USTM(ims,j),CK(ims,j),CKA(ims,j),                  &
                 CD(ims,j),CDA(ims,j),dxCell(ims,j)                 &
 #elif ( EM_CORE == 1)
                 ,isftcflx,iz0tlnd,scm_force_flux,                  &
                 USTM(ims,j),CK(ims,j),CKA(ims,j),                  &
-                CD(ims,j),CDA(ims,j)                               & 
+                CD(ims,j),CDA(ims,j)                               &
```

Comment:
It looks like this could be changed so that the only difference between mpas and EM_CORE is dxCell.

Reply:
@weiwangncar 
I think that EM_CORE (1) does not equate wrf_model. If that's true, then there are 3 categories to deal with here, and since MPAS and EM_CORE both need all of these arguments, I think I have to do it this way so that they each get all their necessary arguments, unless you'd rather have something like:

#if (defined(mpas) || (EM_CORE == 1))
,isftcflx,iz0tlnd,scm_force_flux,                  &
 USTM(ims,j),CK(ims,j),CKA(ims,j),                  &
 CD(ims,j),CDA(ims,j)                          &
#endif
#if defined(MPAS)
,dxCell(ims, j)
#endif

Reply:
@kkeene44 You are right about this. The other way you proposed to do this is more understandable, but it's not necessarily cleaner. Does anyone else have an opinion on this?

Reply:
@weiwangncar 
The code has now been updated to get around this problem. DX is now a 2d array in the subroutine sfclay1D for both WRF/MPAS (which is what we want in the long run anyway). This allows us to only need a 'defined(mpas)' at the top of the subroutine sfclay. All mods compile and run fine for WRF and MPAS, and I have updated the code in this PR with the additional changes.